frontend/src/common/reducers/domainData.js

```diff
@@ -36,11 +38,46 @@ const domainDataReducer = (state = initialDomainDataState, action) => {
         case SET_USER_PROJECT: {
             const settings = {
                 projects: {
+                    [action.project.id]: { $auto: {
+                        $merge: action.project,
+                    } },
+                },
+            };
+
+            if (action.userId) {
+                settings.users = {
+                    [action.userId]: { $auto: {
+                        projects: { $autoArray: {
+                            $push: [action.project.id],
```

Comment:
Check for duplication of projectId in user's projects

Reply:
done